rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java

```diff
@@ -149,44 +153,61 @@ public long getSize() {
      */
     @Override
     public ImportResult tryToConnect(Block block) {
-        if (block == null) {
-            return ImportResult.INVALID_BLOCK;
-        }
+        this.lock.readLock().lock();
 
-        if (!block.isSealed()) {
-            panicProcessor.panic("unsealedblock", String.format("Unsealed block %s %s", block.getNumber(), block.getHash()));
-            block.seal();
-        }
+        try {
+            if (block == null) {
+                return ImportResult.INVALID_BLOCK;
+            }
 
-        if (blockRecorder != null) {
-            blockRecorder.writeBlock(block);
-        }
+            if (!block.isSealed()) {
+                panicProcessor.panic("unsealedblock", String.format("Unsealed block %s %s", block.getNumber(), block.getHash()));
+                block.seal();
+            }
 
-        try {
-            logger.trace("Try connect block hash: {}, number: {}",
-                         block.getShortHash(),
-                         block.getNumber());
-
-            synchronized (connectLock) {
-                logger.trace("Start try connect");
-                long saveTime = System.nanoTime();
-                ImportResult result = internalTryToConnect(block);
-                long totalTime = System.nanoTime() - saveTime;
-                logger.info("block: num: [{}] hash: [{}], processed after: [{}]nano, result {}", block.getNumber(), block.getShortHash(), totalTime, result);
-                return result;
+            if (blockRecorder != null) {
+                blockRecorder.writeBlock(block);
+            }
+
+            try {
+                logger.trace("Try connect block hash: {}, number: {}",
+                        block.getShortHash(),
+                        block.getNumber());
+
+                synchronized (connectLock) {
+                    logger.trace("Start try connect");
+                    long saveTime = System.nanoTime();
+                    ImportResult result = internalTryToConnect(block);
+                    long totalTime = System.nanoTime() - saveTime;
+                    logger.info("block: num: [{}] hash: [{}], processed after: [{}]nano, result {}", block.getNumber(), block.getShortHash(), totalTime, result);
+                    return result;
+                }
+            } catch (Throwable t) {
+                logger.error("Unexpected error: ", t);
+                return ImportResult.INVALID_BLOCK;
             }
-        } catch (Throwable t) {
-            logger.error("Unexpected error: ", t);
-            return ImportResult.INVALID_BLOCK;
+        }
+        finally {
+            this.lock.readLock().unlock();
         }
     }
 
+    @Override
+    public void suspendProcess() {
```

Comment:
Please document and update function name. Why suspend does take the lock and resume unlocks it?